Tests/Resources/app/config/config_test.yml

```diff
@@ -26,3 +26,4 @@ httplug:
 services:
     app.http.plugin.custom:
         class: Http\Client\Common\Plugin\RedirectPlugin
+        public: true
```

Comment:
why do the plugins need to be public? that smells like an issue in how we wire them into the plugin-client, no?

Reply:
Isn't just because test assertions are written against the compiled container instead of the container builder?

Reply:
indeed, seems the problem is in ServiceInstantiationTest::testProfilingShouldNotChangeServiceReference where we get the container from the kernel and then assert that nothing messes with the plugin service. i don't know if there is an easy solution for this - maybe simply add a comment here to avoid confusion? `# plugin services usually do not need to be public. this one is made public so that we can do functional tests on the service`

Reply:
That's basically the approach we still need to document with now "private by default" services in Symfony 4 (see symfony/symfony-docs#8097). In this case, we do not need to work around with an alias.

Reply:
Are there any preferences as to whether I should use a test alias or just add a comment? I personally prefer the comment. Making this public for tests because we want to get it from the container feels more honest to me.

Reply:
The alias IMO is only needed when the service would be present outside the test too (which is not the case here).

Reply:
agreed, lets just add a comment